kyo-stm/shared/src/main/scala/kyo/TChunk.scala

```diff
@@ -0,0 +1,212 @@
+        TRef.init(Chunk.empty[A])
+
+    /** Creates a new transactional chunk within an STM transaction.
+      *
+      * @param chunk
+      *   The initial chunk to wrap
+      * @return
+      *   A new transactional chunk containing the values, within the STM effect
+      */
+    def init[A](values: A*)(using Frame): TChunk[A] < STM = TRef.init(Chunk.from(values))
+
+    /** Creates a new transactional chunk within an STM transaction.
+      *
+      * @param chunk
+      *   The initial chunk to wrap
+      * @return
+      *   A new transactional chunk containing the values, within the STM effect
+      */
+    def init[A](chunk: Chunk[A])(using Frame): TChunk[A] < STM = TRef.init(chunk)
+
+    /** Creates a new empty TChunk outside of a transaction.
+      *
+      * WARNING: This operation:
+      *   - Cannot be rolled back
+      *   - Is not part of any transaction
+      *   - Will cause any containing transaction to retry if used within one, since it creates a reference with a newer transaction ID
+      *
+      * Use this only for static initialization or when you specifically need non-transactional creation. For most cases, prefer `init`.
+      *
+      * @param values
+      *   the initial values to store in the chunk
+      * @return
+      *   a new transactional chunk containing the values within IO
+      */
+    def initNow[A](using Frame): TChunk[A] < IO = TRef.initNow(Chunk.empty)
+
+    /** Creates a new TChunk outside of a transaction.
+      *
+      * WARNING: This operation:
+      *   - Cannot be rolled back
+      *   - Is not part of any transaction
+      *   - Will cause any containing transaction to retry if used within one, since it creates a reference with a newer transaction ID
+      *
+      * Use this only for static initialization or when you specifically need non-transactional creation. For most cases, prefer `init`.
+      *
+      * @param values
+      *   the initial values to store in the chunk
+      * @return
+      *   a new transactional chunk containing the values within IO
+      */
+    def initNow[A](values: A*)(using Frame): TChunk[A] < IO = TRef.initNow(Chunk.from(values))
+
+    /** Creates a new transactional chunk outside of any transaction.
+      *
+      * WARNING: This operation:
+      *   - Cannot be rolled back
+      *   - Is not part of any transaction
+      *   - Will cause any containing transaction to retry if used within one, since it creates a reference with a newer transaction ID
+      *
+      * Use this only for static initialization or when you specifically need non-transactional creation. For most cases, prefer `init`.
+      *
+      * @param chunk
+      *   The initial chunk to wrap
+      * @return
+      *   A new transactional chunk containing the values, within the IO effect
+      */
+    def initNow[A](chunk: Chunk[A])(using Frame): TChunk[A] < IO = TRef.initNow(chunk)
+
+    extension [A](self: TChunk[A])
```

Comment:
Do we need a `use` method?

Reply:
The `use` in `TMap` provides key access. Would it be indexed-based for a `Chunk`? It doesn't seem very safe.

Reply:
My suggestion, was simply `def use[B](f: Chunk[A] => B)` to enable accessing non-exposed methods.

Reply:
Thanks for the clarification! Updated